src/tests/server/database/user.test.js

```diff
@@ -1,92 +1,133 @@
-import test, { after, before, describe } from "node:test";
+import test, { after, afterEach, before, describe } from "node:test";
 import assert from "node:assert/strict";
 import { resolve, join } from "node:path";
+import { randomUUID } from "node:crypto";
 import * as User from "../../../server/database/user.js";
 import {
   initTestDatabase,
   concludeTesting,
+  clearTestData,
   Models,
 } from "../../../server/database/index.js";
 import { ROOT_DIR } from "../../../helpers.js";
 import dotenv from "@dotenvx/dotenvx";
 const envPath = resolve(join(ROOT_DIR, `.env`));
 dotenv.config({ quiet: true, path: envPath });
 
+/**
+ * @typedef {Object} User
+ * @property {number} id
+ * @property {string} name
+ * @property {string} slug
+ * @property {string} created_at
+ * @property {string} enabled_at
+ * @property {string} bio
+ */
+
+/**
+ * Creates a user with a random, unique name
+ *
+ * @returns {User} the created user
+ */
+const createUser = () => {
+  const username = randomUUID();
+  const user = Models.User.create({ name: username });
+
+  return user;
+};
+
+/**
+ * Creates an admin user
+ *
+ * @returns {User} the admin user
+ */
+const createAdminUser = () => {
+  const user = createUser();
+
+  Models.Admin.create({ user_id: user.id });
+
+  return user;
+};
+
 describe(`user tests`, async () => {
   before(async () => await initTestDatabase());
+  afterEach(() => clearTestData());
   after(() => concludeTesting());
 
   /*
   export function processUserLogin(userObject) {
   */
 
   test(`deleteUser`, () => {
-    const user = Models.User.create({ name: `disposable` });
+    const user = createUser();
     User.deleteUser(user);
   });
 
   test(`enable/disable`, () => {
-    let user = User.getUser(`test-user`);
+    let user = createUser();
+    const username = user.name;
 
     User.enableUser(user);
-    user = User.getUser(`test-user`);
+    user = User.getUser(username);
     assert.notEqual(user.enabled_at, null);
 
     User.disableUser(user);
-    user = User.getUser(`test-user`);
+    user = User.getUser(username);
```

Comment:
Note that `getUser` (and the related `getProject`) takes either a slug or id. The user/project name is effectively a cosmetic value, only used in page contexts. The original code uses the slug, but since we're making an anonymous user here, using the id makes more sense:


```suggestion
    let user = createUser();

    User.enableUser(user);
    user = User.getUser(user.id);
    assert.notEqual(user.enabled_at, null);

    User.disableUser(user);
    user = User.getUser(user.id);
```